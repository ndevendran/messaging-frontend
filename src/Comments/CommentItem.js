import React from 'react';

export default ({ comment, commentId, user }) => {
  return (
    <div className="comment">
      <div className="header">
        {user.username}
      </div>
      <div className="body">
        {comment.text}
      </div>
      <div className="footer"></div>
    </div>
  );
}
