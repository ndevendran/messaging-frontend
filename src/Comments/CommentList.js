import React from 'react';
import ResponseItem from '../Common/ResponseItem.js';
import { Query } from 'react-apollo';
import gql from 'graphql-tag';

const GET_COMMENTS = gql`
  query($messageId: ID!) {
    comments(messageId: $messageId) {
      edges {
        id
        createdAt
        text
        user {
          id
          username
        }
      }
    }
  }
`;



export default({ messageId }) => {
  return (
      <Query query={GET_COMMENTS} variables={{ messageId: messageId }}>
        {
          ({ data, loading, error }) => {

            if(loading || !data) {
              return (
                <div>Loading...</div>
              );
            }

            const comments = data.comments.edges;
            return (
              comments.map((comment) =>
                <div key={comment.id}>
                  <ResponseItem response={comment} />
                </div>
              )
            );
          }
        }
      </Query>
  );
}
