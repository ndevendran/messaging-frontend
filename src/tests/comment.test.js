import React from 'react';
import CommentList from '../Comments/CommentList';
import CommentItem from '../Comments/CommentItem';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

Enzyme.configure({adapter: new Adapter()});


describe('Comments', () => {
  const comments = [
    {
      id: 1,
      text: 'Test comment 1',
      user: {
        username: 'lilniro',
      },
    },
    {
      id: 2,
      text: 'Test comment 2',
      user: {
        username: 'wendy666',
      },
    },
  ];

  describe('CommentList', () => {
      it('loads without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<CommentList comments={comments} />, div);
        ReactDOM.unmountComponentAtNode(div);
      });

      it('has a valid snapshot', () => {
        const component = renderer.create(<CommentList comments={comments} />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });

      it('has the correct number of comments', () => {
        const element = shallow(<CommentList comments={comments} />);

        expect(element.find(CommentItem).length).toBe(2);
      });
  });

  describe('CommentItem', () => {
    it('loads without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(<CommentItem comment={comments[1]} />, div);
      ReactDOM.unmountComponentAtNode(div);
    });

    it('has a valid snapshot', () => {
      const div = document.createElement('div');
      const component = renderer.create(<CommentItem comment={comments[1]} />);
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('has a header, body, and footer all under a comment class', () => {
      const element = shallow(<CommentItem comment={comments[0]} />);
      expect(element.find('.comment').length).toBe(1);
      expect(element.find('.header').length).toBe(1);
      expect(element.find('.body').length).toBe(1);
      expect(element.find('.footer').length).toBe(1);
    });
  });
});
