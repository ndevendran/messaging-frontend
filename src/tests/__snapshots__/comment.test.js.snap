// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comments CommentItem has a valid snapshot 1`] = `
<div
  className="comment"
>
  <div
    className="header"
  >
    wendy666
  </div>
  <div
    className="body"
  >
    Test comment 2
  </div>
  <div
    className="footer"
  />
</div>
`;

exports[`Comments CommentList has a valid snapshot 1`] = `
Array [
  <div>
    <div
      className="comment"
    >
      <div
        className="header"
      >
        lilniro
      </div>
      <div
        className="body"
      >
        Test comment 1
      </div>
      <div
        className="footer"
      />
    </div>
  </div>,
  <div>
    <div
      className="comment"
    >
      <div
        className="header"
      >
        wendy666
      </div>
      <div
        className="body"
      >
        Test comment 2
      </div>
      <div
        className="footer"
      />
    </div>
  </div>,
]
`;
